pkg/cmd/develop/cmd.go

```diff
@@ -86,38 +101,95 @@ func NewCmd() *cobra.Command {
+	developCmd.PersistentFlags().StringP("session", "s", "", "create or join an existing session")
+	developCmd.PersistentFlags().StringP("route", "", "", "specifies traffic route options in the format of type:name=value. "+
 		"Defaults to X-Workspace-Route header with current session name value")
-	developCmd.Flags().StringP("namespace", "n", "", "target namespace to develop against "+
+	developCmd.PersistentFlags().StringP("namespace", "n", "", "target namespace to develop against "+
 		"(defaults to default for the current context)")
 
 	developCmd.Flags().VisitAll(config.BindFullyQualifiedFlag(developCmd))
 
-	_ = developCmd.MarkFlagRequired("deployment")
-	_ = developCmd.MarkFlagRequired(execute.RunFlagName)
+	_ = developCmd.MarkPersistentFlagRequired("deployment")
+	_ = developCmd.MarkPersistentFlagRequired(execute.RunFlagName)
 
 	return developCmd
 }
+
+func createDevelopNewCmd() *cobra.Command {
+	newCmd := &cobra.Command{
+		Use:         "new",
+		Short:       "Enables development flow for non-existing service.",
+		Annotations: tpAnnotations,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			name := cmd.Flag("name").Value.String()
+			e := cmd.Parent().PersistentFlags().Set("deployment", name+"-v1")
+
+			return errors.Wrapf(e, "failed populating flags")
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// service-name (provided by --name flag or autogenerated?)
+			// --type deployment|deploymentconfig
+			ns := cmd.Flag("namespace").Value.String()
+			client, err := dynclient.NewDefaultDynamicClient(ns, true)
+			if err != nil {
+				return errors.Wrap(err, "failed creating dynamic client")
+			}
+
+			serviceName := cmd.Flag("name").Value.String()
+
+			var collectedErrors error
+			basicNewService(serviceName, ns, func(object runtime.Object) {
+				creationErr := client.Create(object) // Create k8s objects on the fly
+				collectedErrors = errors.Append(collectedErrors, creationErr)
+			})
+
+			if collectedErrors != nil {
+				fmt.Println(">>>> WE HAVE ERRORS")
+				// FIX should return collectedErrors here but fails with admission webhook - investigate
+			}
+
+			return errors.Wrapf(cmd.Parent().RunE(cmd, args), "failed executing `ike develop` command from `ike develop new`")
+		},
+	}
+
+	newCmd.Flags().String("name", "", "defines service/deployment name")
+	newCmd.Flags().String("type", "deployment", "deployment/deploymentconfig")
+
+	return newCmd
+}
+
+func basicNewService(name, ns string, printer generator.Printer) {
+	newService := generator.NewServiceEntry(name, ns, "Deployment")
+
+	generator.Generate(
+		printer,
+		[]generator.ServiceEntry{newService},
+		generator.AllSubGenerators,
+		generator.WithVersion("v1"),
+		generator.UsingImage("quay.io/maistra-dev/istio-workspace-test-prepared-prepared-image"),
+		generator.ForService(newService, generator.ConnectToGateway(generator.GatewayHost)),
```

Comment:
If I'm not mistaken, this will generate a Gateway CR as well. We probably need some way to avoid that.
https://github.com/maistra/istio-workspace/pull/1050/files#diff-165df1771e0e1d8ede00de7b3c127f05aebf9e939a933a8ebb0d50e4421ef427R82

Reply:
Yeah, that's the reason why I'm seeing errors from admission webhook. It's in TODO. Thanks for pointing it out.

Reply:
Fixed with fb5a2da